### PR TITLE
tools: display PID intead of TID in statsnoop.py

### LINKED ISSUE #3332: statsnoop not supported PID
```
# statsnoop
63764  ForkJoinPool-3-     0   0 /etc/resolv.conf
63205  ForkJoinPool-3-     0   0 /etc/resolv.conf
25986  ForkJoinPool-3-     0   0 /etc/resolv.conf
63236  ForkJoinPool-3-     0   0 /etc/resolv.conf
63252  ForkJoinPool-3-     0   0 /etc/resolv.conf
63230  ForkJoinPool-3-     0   0 /etc/resolv.conf
63777  ForkJoinPool-3-     0   0 /etc/resolv.conf
```

nothing when set pid
```
# ps -eT|grep 63764
61129 63764 ?        00:00:03 ForkJoinPool-3-

# statsnoop -p 61129
PID    COMM               FD ERR PATH
^C
```

```
# rpm -qa |grep bcc
python-bcc-0.15.0-2.el7.noarch
bcc-0.15.0-2.el7.x86_64
bcc-tools-0.15.0-2.el7.x86_64
```

### PR DESCRIPTION
**Pull request by chenhengqi**
This PR is a first step to #3407 and fix #3332 .

Signed-off-by: Hengqi Chen <chenhengqi@outlook.com>

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by chenhengqi** [comment]
@yonghong-song 

I have several updates to this tool, please review again.

* remove an unused HASH_MAP
* use userspace PID in output
* enable the `failed` option
* adjust `PID` column's width

**Comment by yonghong-song** [comment]
[buildbot, test this please]

File: tools/statsnoop.py [hunk 1]
```diff
@@ -56,42 +56,41 @@
...
-BPF_HASH(infotmp, u32, struct val_t);
+BPF_HASH(infotmp, u64, struct val_t);
 BPF_PERF_OUTPUT(events);
 
 int syscall__entry(struct pt_regs *ctx, const char __user *filename)
 {
     struct val_t val = {};
-    u32 pid = bpf_get_current_pid_tgid();
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u32 pid = pid_tgid >> 32;
```
Comment (yonghong-song) [inline tools/statsnoop.py:66]: Using `tid = bpf_get_current_pid_tgid();` totally fine as the key although using pid_tgid works too. Let us just maintain the original code but rename "pid" to "tid". In some other bcc tools, people use tid too. Let us try to get consistent about this. Something like
```
  u64 pid_tgid = bpf_get_current_pid_tgid();
  u32 pid = pid_tgid >> 32, tid = (u32)pid_tgid;
```

File: tools/statsnoop.py [hunk 2]
```diff
@@ -56,42 +56,41 @@
     char fname[NAME_MAX];
 };
 
-BPF_HASH(args_filename, u32, const char *);
-BPF_HASH(infotmp, u32, struct val_t);
+BPF_HASH(infotmp, u64, struct val_t);
```
Comment (yonghong-song) [inline tools/statsnoop.py:59]: Let us keep this as u32, see below.

File: tools/statsnoop.py [hunk 3]
```diff
@@ -56,42 +56,41 @@
...
 
-    valp = infotmp.lookup(&pid);
+    valp = infotmp.lookup(&pid_tgid);
     if (valp == 0) {
         // missed entry
         return 0;
     }
 
-    struct data_t data = {.pid = pid};
+    struct data_t data = {.pid = pid_tgid >> 32};
```
Comment (yonghong-song) [inline tools/statsnoop.py:86]: Yes, this is the place you want to change.

**Review by yonghong-song — COMMENTED** [review]
Could you change the subject line to "display PID intead of TID in statsnoop.py"? Thanks!

File: tools/statsnoop.py (hunk 1 above)
Comment (chenhengqi) [reply tools/statsnoop.py:66]: Ah, tid is unique across all processes, right ?

**Comment by brendangregg** [comment]
Why did the PID  column become so much bigger?

```
 print("%-10s %-16s %4s %3s %s" % ("PID", "COMM", "FD", "ERR", "PATH"))
```

I make the output as narrow as possible so it fits on slides (which may have a 40-60 character width) and books (75-85 character width).

And please test these tools: the "PID" it reports should be the same "PID" you see in `ps -ef` and `pgrep`.

**Comment by chenhengqi** [comment]
> Why did the PID column become so much bigger?
> 
> ```
>  print("%-10s %-16s %4s %3s %s" % ("PID", "COMM", "FD", "ERR", "PATH"))
> ```
> 
> I make the output as narrow as possible so it fits on slides (which may have a 40-60 character width) and books (75-85 character width).
> 
> And please test these tools: the "PID" it reports should be the same "PID" you see in `ps -ef` and `pgrep`.

1. the Ubuntu distro has its default `pid_max` set to 4194304, so 6 is not enough, maybe I should change it to 8 ? Or I can drop this change if it is not appropriate.
2. The PID field is tested with those tools, as you mentioned in #3407

**Comment by brendangregg** [comment]
Ah, thanks. pid_max of 4194304 is kinda ugly, but I guess they are doing it to support containers.

I'd make it %7d. I don't want to set a precedent for anything bigger, as in other tools we need every single character we have to fit within 75-80.

**Comment by chenhengqi** [comment]
Updated.

File: tools/statsnoop.py [hunk 4]
```diff
@@ -56,42 +56,43 @@
...
+    infotmp.update(&tid, &val);
 
     return 0;
 };
 
 int trace_return(struct pt_regs *ctx)
 {
-    u32 pid = bpf_get_current_pid_tgid();
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u32 tid = pid_tgid & 0xFFFFFFFF;
```
Comment (yonghong-song) [inline tools/statsnoop.py:79]: Looks good. Thanks. Let us use `u32 tid = (u32)pid_tgid;` similar to the above.

Comment (chenhengqi) [reply tools/statsnoop.py]: Done.

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request merged**